workloads/imagenet/imagenet_jax/workload.py

```diff
@@ -0,0 +1,314 @@
+"""ImageNet workload implemented in Jax."""
+
+from typing import Tuple, Any, Callable
+import time
+import functools
+from absl import logging
+
+import tensorflow as tf
+# Hide any GPUs form TensorFlow. Otherwise TF might reserve memory and make it
+# unavailable to JAX.
+tf.config.experimental.set_visible_devices([], 'GPU')
+import tensorflow_datasets as tfds
+
+import jax
+import jax.numpy as jnp
+from jax import lax
+import flax
+from flax import optim
+from flax.training import common_utils
+
+import spec
+from . import input_pipeline
+from . import models
+from . import submission
+from . import config as config_lib
+
+config = config_lib.get_config()
+
+
+
+# flax.struct.dataclass enables instances of this class to be passed into jax
+# transformations like tree_map and pmap.
+@flax.struct.dataclass
+class TrainState:
+  step: int
+  optimizer: optim.Optimizer
+  model_state: Any
+
+
+class ImagenetWorkload(spec.Workload):
+  def __init__(self):
+    self._eval_ds = None
+    self._param_shapes = None
+    self.epoch_metrics = []
+
+  def has_reached_goal(self, eval_result: float) -> bool:
+    return eval_result > 0.6
+
+  def _build_dataset(self,
+      data_rng: jax.random.PRNGKey,
+      split: str,
+      data_dir: str,
+      batch_size):
+    if config.batch_size % jax.device_count() > 0:
+      raise ValueError('Batch size must be divisible by the number of devices')
+    local_batch_size = config.batch_size // jax.host_count()
+
+    if config.half_precision:
+      if platform == 'tpu':
+        input_dtype = tf.bfloat16
+      else:
+        input_dtype = tf.float16
+    else:
+      input_dtype = tf.float32
+
+    ds_builder = tfds.builder(config.dataset)
+    ds = input_pipeline.create_input_iter(
+      ds_builder,
+      local_batch_size,
+      config.image_size,
+      input_dtype,
+      train=split=='train',
+      cache=config.cache)
```

Comment:
I would hardcode this to True instead, I believe it should always improve performance